### PR TITLE
feat: improve notifications activity

### PR DESCRIPTION
**Pull request by TheGuyWithoutH**
- Add redirect to pages when clicking on notifications
- Fix like update
- Add profile pictures follow notification

### CODE REVIEW

File: app/src/main/java/ch/epfl/culturequest/social/PictureAdapter.java [hunk 1]
```diff
@@ -179,8 +179,9 @@ private void handleLike(@NonNull PictureViewHolder holder, Post post) {
             if (holder.isLiked) {
                 holder.isLiked = false;
                 Database.removeLike(post, Profile.getActiveProfile().getUid()).whenComplete((aVoid, throwable) -> {
-                    if (throwable != null) {
+                    if (throwable == null) {
```
Comment (hmiranda-queiros) [inline app/src/main/java/ch/epfl/culturequest/social/PictureAdapter.java:182]: maybe you could also test if "aVoid" is not null ?

File: app/src/main/java/ch/epfl/culturequest/ui/notifications/NotificationsRecycleViewAdapter.java [hunk 2]
```diff
@@ -42,11 +47,42 @@ public NotificationsRecycleViewAdapter.NotificationViewHolder onCreateViewHolder
...
             Database.deleteNotification(Profile.getActiveProfile().getUid(), notificationTexts.get(position));
             notificationTexts.remove(position);
             notifyItemRemoved(position);
+            notifyItemRangeChanged(0, getItemCount());
+        });
+
+        holder.itemView.setOnClickListener(view -> {
+            System.out.println(notificationTexts.get(position).getChannelId());
+            switch (notificationTexts.get(position).getChannelId()) {
+                case ScanNotification.CHANNEL_ID:
```
Comment (hmiranda-queiros) [inline app/src/main/java/ch/epfl/culturequest/ui/notifications/NotificationsRecycleViewAdapter.java:62]: Why don't you use "notificationTexts.get(position).buildnotification().getPendingIntent().send();" instead of coding it again here ?

**Review by hmiranda-queiros — COMMENTED** [review]
Great work, the notification activity should be more robust now !
I just put some comments on possible improvements :)

File: app/src/main/java/ch/epfl/culturequest/ui/notifications/NotificationsRecycleViewAdapter.java (hunk 2 above)
Comment (TheGuyWithoutH) [reply app/src/main/java/ch/epfl/culturequest/ui/notifications/NotificationsRecycleViewAdapter.java:62]: I cannot find getPendingIntent, are you sure the Notification class implement this method?

**Review by hmiranda-queiros — APPROVED** [review]
All good to go !
